Use `PortalContext` in MUI components (#1762)
iTwin/stratakit: apps/website/src/content/docs/components/mui/Popover.md

- Added [`role="dialog"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/dialog_role) to the [`paper`](https://mui.com/material-ui/api/popover/#Popover-css-MuiPopover-paper) slot.
- Added fallback mechanism for automatically labelling the `paper` slot using the trigger element (`anchorEl`).
- `disableScrollLock` is used to prevent scroll locking when the popover is open.

> - The default [`container`](https://mui.com/material-ui/api/modal/#modal-prop-container) now resolves from StrataKit's portal context, which defaults to [`portalContainer`](https://stratakit.bentley.com/docs/components/root/#portal-container).

Reviewer: What is `portalContainer`? The linked section in `/root` docs only explains the general concept of "root portal container".

The old link text + local URL seems more appropriate.

Also, it should point to the most specific API docs ([`Popover.container`](https://mui.com/material-ui/api/popover/#popover-prop-container) in this case).

```suggestion
- The default [`container`](https://mui.com/material-ui/api/popover/#popover-prop-container) now resolves from StrataKit's portal context, which defaults to [root portal container](/components/root/#portal-container).
```

(similar applies to other instances)

> - Nested elements are now portaled into the container specified by the portal provider.

Reviewer: "portal provider" is not a previously established concept.

```suggestion
- Nested elements now use the Popover as the portal container.
```

(similar wording can be used for Dialog and Drawer)
